Tallies/TallyClerks/MGxsClerk_class.f90

```diff
@@ -0,0 +1,887 @@
+module MGxsClerk_class
+
+  use numPrecision
+  use tallyCodes
+  use endfConstants
+  use genericProcedures,          only : fatalError
+  use dictionary_class,           only : dictionary
+  use particle_class,             only : particle, particleState
+  use particleDungeon_class,      only : particleDungeon
+  use outputFile_class,           only : outputFile
+
+  ! Nuclear Data interface
+  use nuclearDatabase_inter,      only : nuclearDatabase
+  use neutronXSPackages_class,    only : neutronMacroXSs
+  use neutronMaterial_inter,      only : neutronMaterial,neutronMaterial_CptrCast
+
+  ! Tally Maps
+  use tallyMap_inter,             only : tallyMap
+  use tallyMapFactory_func,       only : new_tallyMap
+
+  ! Tally Interfaces
+  use scoreMemory_class,          only : scoreMemory
+  use tallyClerk_inter,           only : tallyClerk, kill_super => kill
+
+  implicit none
+  private
+
+  !! Size of clerk memory
+  integer(shortInt), parameter  :: ARRAY_SCORE_SIZE   = 7 ,&  ! Size of data to store as 1D arrays
+                                   MATRIX_SCORE_SMALL = 3 ,&  ! Size of data to store as 2D arrays when scoring up to P1
+                                   MATRIX_SCORE_FULL  = 9     ! Size of data to store as 2D arrays when scoring up to P7
+
+  !! Locations of different bins wrt memory address of the clerk
+  integer(longInt), parameter   :: FLUX_idx      = 1 ,&  ! Flux
+                                   SCATT_idx     = 2 ,&  ! Scattering macroscopic reaction rate
+                                   CAPT_idx      = 3 ,&  ! Capture macroscopic reaction rate
+                                   FISS_idx      = 4 ,&  ! Fission macroscopic reaction rate
+                                   NUBAR_idx     = 5 ,&  ! NuBar
+                                   CHI_idx       = 6 ,&  ! Fission neutron spectrum
+                                   SCATT_EV_idx  = 7     ! Analog: number of scattering events
+
+  !!
+  !! Multi-group macroscopic cross section calculation
+  !!
+  !! It prints out:
+  !! capture xs, fission xs, transport xs, nu, chi, the P0 and P1 scattering matrices,
+  !! the P0 scattering production matrix. On request, also the P2 -> P7 scattering matrices.
```

Comment:
Is there any particular reason why the option between P1 and P7 is binary? 
I mean it should be possible for the user to specify the highest moment which is required. I don't think it needs to be a part of this PR necessary but I would be inclined to open an issue about letting N be user-defined parameter after the PR is merged.

Reply:
It was mainly laziness, inspired by Serpent but with an additional high moments-low moments check

Reply:
We will just need to open an issue about it sice I imagine that being able to reduce the number of tallied moments could help with memory consumption.